specs/merge/beacon-chain.md

```diff
@@ -168,7 +168,9 @@ def process_block(state: BeaconState, block: BeaconBlock) -> None:
     process_randao(state, block.body)
     process_eth1_data(state, block.body)
     process_operations(state, block.body)
-    process_execution_payload(state, block.body)  # [New in Merge]
+    # Pre-merge, skip execution payload processing
+    if is_transition_completed(state) or is_transition_block(state, block):
+        process_execution_payload(state, block.body.execution_payload)  # [New in Merge]
```

Comment:
Two issues:
1. Looking at `tests/formats/operations/README.md` and I just realized that we put `sync_aggregate` as part of "operation" tests but `process_sync_committee` is not part of the `process_operations` function like `execution_payload`.
    1. IMO it's good to consider `sync_aggregate` as an operation.
    2. If we want to simplify it, I think `execution_payload` could also be considered as an operation since it's essentially "transactions".
2. I can see why it looks good to extract the `is_transition_completed(state) or is_transition_block(state, block)` condition checks from `process_execution_payload`, but it makes it a bit unclear where/how to test these conditions in test cases.

What do you think about:
1. Move `process_sync_committee` and `process_execution_payload` to `process_operations`.
2. Revert the `if is_transition_completed(state) or is_transition_block(state, block)` checks extraction.

Reply:
1. Regarding `process_sync_committee`: sure, but let's keep that separate from merge-testing for now. Moving the `process_execution_payload` to `process_operations` would be good too. I'm honestly surprised we didn't do the same for the Randao and Eth1Data functionality.
2. The if-statement was purposefully extracted from the `process_execution_payload` function: this way the function input can just be an `ExecutionPayload`, and not a complete beacon block body. This helps with future compatibility and reduces test complexity (We don't have to set up a complete beacon block for every single test case). We can test that one if-statement as part of the regular full beacon block testing. And in future phases after the merge, we may even remove the if-statement entirely, since the merge-transition check will always be true.